sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/pool/SerializableSessionPool.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.jms.pool;
+
+import java.io.Closeable;
+import java.io.Serializable;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.commons.pool2.ObjectPool;
+
+public abstract class SerializableSessionPool<T> implements ObjectPool<T>, Serializable, Closeable {
```

Comment:
Can we add some tests for this session pool in specific? It seems like there is a theoretical risk of incorrectly borrowing/returning objects

Reply:
I did some tests using a Docker image of `SolaceUI` and closing the connection manually. Everything works fine. I tried to add some Unit Tests but I was blocked. I was thinking to pass a SessionPoolFactory so that it makes the tests easier to write. What do you think?

Reply:
I'm all for writing coed that is easy to test